### DISCUSSION TITLE
Uno Linux Problem upgrading VisualStudioCode and .Net

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#16701] [Q&A]
I had a project which was building and running fine with Vscode 1.75 and .Net Core 7.0

I upgraded vscode to latest version 1.79 and installed .NetCore 8. I did not delete .Net core 7.

Now, when I try to build the application, the following error occurs

```
MSBuild version 17.9.8+b34f75857 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
CSC : error UXAML0001: Invalid URI: The URI is empty. [/home/howard/Desktop/SudokuUnoTestNet7/SudokuUnoTestNet7/SudokuUnoTestNet7.Skia.Gtk/SudokuUnoTestNet7.Skia.Gtk.csproj]
```

What does this error mean? The Skia,Gtk.csproj file exists and is unchanged

I have changed nothing else.

I get the same error if I try to build the application after changing all reference to .net7.0 to .net8.0 per the web page on how to upgrade from .Net7.0 to .Net 8.0

Here is the .csproj file

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType Condition="'$(Configuration)'=='Release'">WinExe</OutputType>
    <OutputType Condition="'$(Configuration)'=='Debug'">Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ApplicationManifest>app.manifest</ApplicationManifest>
  </PropertyGroup>
  <ItemGroup Condition="exists('..\SudokuUnoTestNet7.UWP')">
    <EmbeddedResource Include="..\SudokuUnoTestNet7.UWP\Package.appxmanifest" LogicalName="Package.appxmanifest" />
    <Content Include="..\SudokuUnoTestNet7.UWP\Assets\StoreLogo.png" Link="Assets\StoreLogo.png" />
  </ItemGroup>
  <ItemGroup>
    <UpToDateCheckInput Include="..\SudokuUnoTestNet7.Shared\**\*.xaml" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.UI.Skia.Gtk" Version="4.9.17" />
    <PackageReference Include="Uno.UI.RemoteControl" Version="4.9.17" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.9.17" />
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>
  <Import Project="..\SudokuUnoTestNet7.Shared\SudokuUnoTestNet7.Shared.projitems" Label="Shared" />
</Project>
```

Howard Weiss

**Comment by HowardPWeiss** [comment]
```
<Project Sdk="Microsoft.NET.Sdk">
<PropertyGroup>
    <OutputType Condition="'$(Configuration)'=='Release'">WinExe</OutputType>
    <OutputType Condition="'$(Configuration)'=='Debug'">Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ApplicationManifest>app.manifest</ApplicationManifest>
  </PropertyGroup>
  <ItemGroup Condition="exists('..\SudokuUnoTestNet7.UWP')">
    <EmbeddedResource Include="..\SudokuUnoTestNet7.UWP\Package.appxmanifest" LogicalName="Package.appxmanifest" />
    <Content Include="..\SudokuUnoTestNet7.UWP\Assets\StoreLogo.png" Link="Assets\StoreLogo.png" />
  </ItemGroup>
  <ItemGroup>
    <UpToDateCheckInput Include="..\SudokuUnoTestNet7.Shared\**\*.xaml" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.UI.Skia.Gtk" Version="4.9.17" />
    <PackageReference Include="Uno.UI.RemoteControl" Version="4.9.17" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.9.17" />
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>
  <Import Project="..\SudokuUnoTestNet7.Shared\SudokuUnoTestNet7.Shared.projitems" Label="Shared" />
</Project>
```

**Comment by jeromelaban** [comment]
Thanks for the report! Does this happen with a new project? If it doesn't, can you share a repro?

**Comment by HowardPWeiss** [comment]
Jerome

I have created a repo and invited you to collaborate on it. I am relatively new to GitHub.  Hopefully, I have done this correctly

I developed this project and it was working before I upgraded dotnet, .net sdk, uno,  and VisualStudioCode.  I have a windows computer on which I can still compile and run the application using Visual Studio 2022.  On that system, I have Dotnet 7.0.304 and .net sdk 7.0.102.  I am unsure how to find a version for Uno itself.

uno-check on the windows system is v1.12.1.0
uno.dll version is 4.3.37+Branch.release.stable-4.7.....

**Reply by jeromelaban** [reply]
Thanks. Is this the intent to stay on an older release of Uno? We won't be able to make changes to it.

**Reply by jeromelaban** [reply]
> Not at all. I have already migrated the application to .Net 8,0. I am in the process of figuring out button styles, visual state manager, etc.

I see, then it's best if you migrate to Uno Platform 5.x, as this particular issue has been fixed in later versions.

**Reply by jeromelaban** [reply]
> Not a problem
> How do I determine the Uno platform version?

You can follow [these upgrading guides](https://platform.uno/docs/articles/migrating-from-previous-releases.html).

**Reply by jeromelaban** [reply]
> How do I determine the Uno platform version?

This is the version of the installed `uno.ui`, `uno.winui` or `uno.sdk` nuget package, depending on the age of your solution.

**Comment by HowardPWeiss** [comment]
Not at all.  I have already migrated the application to .Net  8,0. I am in the process of figuring out button styles, visual state manager, etc. 

My question was: What did the error mean? I did follow the upgrade instructions to switch from .Net 7 to .Net 8, so I expected that the application would compile.

**Reply by HowardPWeiss** [reply]
Not a problem

How do I determine the Uno platform version?

**Comment by HowardPWeiss** [comment]
Thank you for the link  to the upgrading guides

How do I determine the current installed version of the Uno Platform?
